zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -462,8 +462,9 @@ public void createNode(final String path, byte[] data, List<ACL> acl, long ephem
             // we did for the global sessions.
             Long acls = aclCache.convertAcls(acl);
 
-            Set<String> children = parent.getChildren();
-            if (children.contains(childName)) {
+            DataNode existingChild = nodes.get(path);
+            if (existingChild != null) {
+                existingChild.acl = acls;
```

Comment:
I have not had a chance to play with this, but: this means we are relying on `PrepRequestProcessor` to prevent `OpCode.create*` operations from being abused as `OpCode.setACL`, right? (Just thinking out loud.)

Reply:
I'm not sure to be honest and not sure if I understand your question.